src/main/java/io/codelex/streams/practice/StreamsExercise.java

```diff
@@ -0,0 +1,128 @@
+package io.codelex.streams.practice;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * TODO - search for StreamsExerciseTest and make all tests pass
+ */
+public class StreamsExercise {
+    public static List<Integer> returnSquareRoot(List<Integer> numbers) {
+        return numbers.stream().map((Integer n) -> (int) Math.sqrt(n)).collect(Collectors.toList());
+    }
+
+    public static List<Integer> getAgeFromUsers(List<User> user) {
+        return user.stream().map(User::getAge).collect(Collectors.toList());
+    }
+
+    public static List<Integer> getDistinctAges(List<User> users) {
+        return users.stream().map(User::getAge).distinct().collect(Collectors.toList());
+    }
+
+    public static List<User> getLimitedUserList(List<User> users, int limit) {
+        return users.stream().limit(limit).collect(Collectors.toList());
+    }
+
+    public static Integer countUsersOlderThen25(List<User> users) {
+        return (int) users.stream().map(User::getAge).filter(age -> age > 25).count();
+    }
+
+    public static List<String> mapToUpperCase(List<String> strings) {
+        return strings.stream().map(String::toUpperCase).collect(Collectors.toList());
+    }
+
+    public static Integer sum(List<Integer> integers) {
+        return integers.stream().mapToInt(Integer::intValue).sum();
+    }
+
+    public static List<Integer> skip(List<Integer> integers, Integer toSkip) {
+        return integers.stream().skip(toSkip).collect(Collectors.toList());
+    }
+
+    public static List<String> getFirstNames(List<String> names) {
+        return names.stream().map((String name) -> name.split(" ")[0]).collect(Collectors.toList());
+    }
+
+    public static List<String> getDistinctLetters(List<String> names) {
+        return names.stream().map((String n) -> n.split("")).flatMap(Arrays::stream).distinct().collect(Collectors.toList());
```

Comment:
for readability, par sliktu nenāk jaunu funkciju jaunā rindā
```
.stream()
.map()
.flatMap()
.collection()

Reply:
Done.